pkg/tool/osdctl/osdctl.go

```diff
@@ -0,0 +1,199 @@
+	}
+
+	// Verify checksum of downloaded assets
+	osdctlBinaryFilepath := filepath.Join(versionedDir, osdctlBinaryAsset.GetName())
+	fileBytes, err := os.ReadFile(osdctlBinaryFilepath)
+	if err != nil {
+		return fmt.Errorf("failed to read osdctl binary file '%s' while generating sha256sum: %w", osdctlBinaryFilepath, err)
+	}
+	sumBytes := sha256.Sum256(fileBytes)
+	// TODO - there's probably a better way to do this
+	binarySum := fmt.Sprintf("%x", sumBytes[:])
+
+	checksumFilePath := filepath.Join(versionedDir, checksumAsset.GetName())
+	checksumBytes, err := os.ReadFile(checksumFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read osdctl checksum file '%s': %w", checksumFilePath, err)
+	}
+	checksum := strings.Split(string(checksumBytes), " ")[0]
+	if strings.TrimSpace(binarySum) != strings.TrimSpace(checksum) {
+		fmt.Printf("WARNING: Checksum for osdctl does not match the calculated value. Please retry installation. If issue persists, this tool can be downloaded manually at %s\n", osdctlBinaryAsset.GetBrowserDownloadURL())
+		// We shouldn't link this binary to latest if the checksum isn't valid
+		return nil
+	}
+
+	if err := unArchive(t.toolDir(""), filepath.Join(versionedDir, checksumAsset.GetName())); err != nil {
+		return fmt.Errorf("failed to unarchive the checksum file '%s': %w", checksumFilePath, err)
+	}
+	//Untar osdctl file and remove (so, it removes .tar.gz basically)
+
+	// Link as latest
+	latestFilePath := t.symlinkPath(latestDir)
+	err = os.Remove(latestFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to remove existing 'osdctl' binary at '%s': %w", latestDir, err)
+	}
+
+	err = os.Symlink(osdctlBinaryFilepath, latestFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to link new 'osdctl' binary to '%s': %w", latestDir, err)
+	}
+	return nil
+}
+
+// toolDir returns this tool's specific directory given the root directory all tools are installed in
+func (t *Tool) toolDir(rootDir string) string {
+	return filepath.Join(rootDir, "osdctl")
+}
+
+func (t *Tool) symlinkPath(latestDir string) string {
+	return filepath.Join(latestDir, "osdctl")
+}
+
+func (t *Tool) Remove(rootDir, latestDir string) error {
+	// Remove all binaries owned by this tool
+	toolDir := t.toolDir(rootDir)
+	err := os.RemoveAll(toolDir)
+	if err != nil {
+		return fmt.Errorf("failed to remove %s: %w", toolDir, err)
+	}
+
+	// Remove all symlinks owned by this tool
+	latestFilePath := t.symlinkPath(latestDir)
+	err = os.Remove(latestFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to remove symlinked file %s: %w", latestFilePath, err)
+	}
+	return nil
+}
+
+func unArchive(source string, destination string) error {
+	src, err := os.Open(source)
+	if err != nil {
+		return nil
+	}
+	defer src.Close()
+	uncompressed, err := gzip.NewReader(src)
+	if err != nil {
+		return nil
+	}
+	arc := tar.NewReader(uncompressed)
+	for {
+		var f *tar.Header
+		f, err = arc.Next()
```

Comment:
Let's either initialize the variable as a part of the `arc.Next()` call
```suggestion
for {
		f, err := arc.Next()
```

or move the declaration outside the for loop:
```suggestion
var f *tar.Header
for {
        f, err = arc.Next()
```